pkg/kubelet/cm/dra/manager.go

```diff
@@ -302,34 +335,58 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 				pluginName = claimInfo.DriverName
 			}
 
-			// Call NodeUnprepareResource RPC for each resourceHandle
-			client, err := dra.NewDRAPluginClient(pluginName)
-			if err != nil {
-				return fmt.Errorf("failed to get DRA Plugin client for plugin name %s, err=%+v", pluginName, err)
+			batches[pluginName] = append(batches[pluginName],
+				&drapb.Claim{
+					Namespace:      resourceClaim.Namespace,
+					Uid:            string(resourceClaim.UID),
+					Name:           resourceClaim.Name,
+					ResourceHandle: resourceHandle.Data,
+				})
```

Comment:
Also same comment as above for including all claims in each underlying call to `NodeUnprepareResources()` (even if some of them have already been unprepared.

Reply:
For prepare, always including all of them makes sense to me.

However, during unprepare we have an additional complication: some claims of the pod might still be in use by other pods. What do we do about those? If we include them, we have to tell the driver "here are some claims, but don't unprepare them". This makes the API more complicated.

I prefer the current approach of only listing claims which really need to be unprepared.

Reply:
Yeah, it doesn't make it as symmetrical but I don't think having symmetry in this case in worth the extra complication. In any case it will get the full list on prepare, so any internal tracking the driver may want to do about whether a pod's full set of claims have been unprepared or not should still be doable.

Reply:
The driver can determine when a pod definitely doesn't use a claim anymore (when `NodeUnprepareResources` is called), but the pod may already have stopped using it already much earlier (because some other pod still used it). That's okay, just wanted to mention it.